src/JustEat.StatsD/EndpointLookups/DnsEndpointProvider.cs

```diff
@@ -10,7 +10,8 @@ public class DnsEndpointProvider : IDnsEndpointMapper
     {
         public IPEndPoint GetIPEndPoint(string hostName, int port)
         {
-            return new IPEndPoint(Dns.GetHostAddresses(hostName)[0], port);
+            var endpoints = Dns.GetHostAddressesAsync(hostName).Result;
```

Comment:
I thought about that, but to properly make it all async means either breaking changes to add new methods to the interfacesand/or change the signatures.

Alternatively we could add new interfaces that support async and implement them where appropriate?